// config/list.js

var mongoose = require('mongoose');
var user = require('config/models');
var subject = require('config/subjects');


//GET - Return all users in the DB
exports.findAllUsers = (req, res) => {
	console.log('GET /users')

  user.find((err, users) => {
	  if (err) return res.send(500, err.message)

	  res.status(200).jsonp(users)

  })

}


//DELETE - Delete a User with specified ID
exports.deleteUser = (req, res) => {
    const id = req.params.id
    console.log(`DELETE /users/${id}`)

    user.findById(id, (err, users) => {

    		if (!user) return res.status(404).send({message: 'User not exist'})

        users.remove( (err) => {
        	if (err) return res.status(500).send(err.message)
      		
      		res.status(200).jsonp({message: 'User deleted'})
        })
    })
}


//GET - Return all  subjects  in the DB
exports.findAllUsersSubj = (req, res) => {
	console.log('GET /users/api')

  subject.find((err, subjects) => {
	  if (err) return res.send(500, err.message)

	  res.status(200).jsonp(subjects)

  })

}

//GET - Return all  subjects with specified ID in the DB
/*exports.findAllUserSubj = (req, res) => {
	const id = req.params.id
	console.log('GET /users/api/${id}')

	subject.findById(id, (err, subjects) => {
  
	  if (err) return res.send(500, err.message)

	  res.status(200).jsonp(subjects)

	})

}*/

//GET - Return all  subjects with specified ID in the DB
exports.findUserSubj = (req, res) => {
	var id = req.params.id
	console.log('POST /users/api/subj/${id}')
	//console.log('GET /users/api/subj/${id}')

	//subject.find({'User' :id}, (err, subjects) => { // Saca todos los campos
	//subject.find({'User' :id}).select(' subj groupte grouppa grouppl profname profemail').exec(function (err, subjects) { 
	subject.find({'User' :id}, {'_id':0,'subj':1,'groupte':1,'grouppa':1,'grouppl':1,'profname':1,'profemail':1}, function(err, subjects){
	//subject.find().populate('User',{where :{token:id}}).exec(function (err, subjects){
	//subject.find().populate({ path: 'User', select: 'token'}).exec(function (err, subjects){
	////subject.find({}).populate({path: 'User',match: { token: id }, select :'id'}).exec(function (err, subjects){
	//subject.find({"User.token":{where:id.match("User.token")}}).exec(function (err, subjects){
	//subject.find({$where:"'id'.match(this.User.token)"}).exec(function (err, subjects){
	//subject.find({}).where('User.token').gt(id).exec(function (err, subjects){
    //subject.find().populate('User').where('token').equals(id).exec(function (err, subjects){
	
  
  
	  if (err) return res.send(500, err.message)

	  res.status(200).jsonp(subjects)
	  

	})
	

}

// GET - Return all users from a specific group of a subject
exports.findUserByPlGroup = (req, res) => {
	var subj = req.params.subj
	var group = req.params.group
	var array = []
	array = group.split(";")
	//var s = subj.replace(/(.)([A-Z])/g, "$1 &2");
	var s = subj.replace(/([A-Z])/g, " $1").trim();
	

	console.log('POST /users/api/subj/${subj}/${group}')
	

	//subject.find({'User' :id}, (err, subjects) => { // Saca todos los campos
	//subject.find({'User' :id}).select(' subj groupte grouppa grouppl profname profemail').exec(function (err, subjects) { 
	//subject.find({'User' :id}, {'_id':0,'subj':1,'groupte':1,'grouppa':1,'grouppl':1,'profname':1,'profemail':1}, function(err, subjects){
	//subject.find().populate('User',{where :{token:id}}).exec(function (err, subjects){
	//subject.find().populate({ path: 'User', select: 'token'}).exec(function (err, subjects){
	////subject.find({}).populate({path: 'User',match: { token: id }, select :'id'}).exec(function (err, subjects){
	//subject.find({"User.token":{where:id.match("User.token")}}).exec(function (err, subjects){
	//subject.find({$where:"'id'.match(this.User.token)"}).exec(function (err, subjects){
	//subject.find({}).where('User.token').gt(id).exec(function (err, subjects){
    //subject.find().populate('User').where('token').equals(id).exec(function (err, subjects){
    
	//subject.find({subj: subj, grouppl:group},{}).populate('User'),function(err,subjects){
	//subject.find({subj: subj, grouppl:group, User :{ $ne: null }},{'_id':0,'subj':0,'groupte':0,'grouppa':0,'grouppl':0,'profname':0,'profemail':0,'__v':0}).where("User").ne(null).populate({path:'User',match:{typeuser:/Alumno/},select:'email -_id'}).exec(function (err, subjects){
	//subject.find({subj:s , grouppl:{ $in: array }, User :{$exists : true, $ne: null }},{'subj':0,'groupte':0,'grouppa':0,'profname':0,'profemail':0,'__v':0}).populate({path:'User',select:'email + typeuser + token'}).exec(function (err, subjects){
	subject.find({subj:s , grouppl:{ $in: array }, User :{$exists : true, $ne: null }},{'subj':0,'groupte':0,'grouppa':0,'profname':0,'profemail':0,'__v':0}).populate({path:'User',select:'email'}).exec(function (err, subjects){
	//subject.find({subj:s , grouppl:{ $in: array }, User :{$exists : true, $ne: null }},{'subj':0,'groupte':0,'grouppa':0,'profname':0,'profemail':0,'__v':0}).populate({path:'User',match:{typeuser:/Alumno/},select:'email -_id'}).exec(function (err, subjects){
	
	//{ $regex: /S/,subj }

	/*subjects = subjects.filter(function( element ) {
	return element !== null;
	});*/

	
	  if (err) return res.send(500, err.message)

	  res.status(200).jsonp(subjects)
	  

	})
	

}


//DELETE - Delete a User subjects with specified ID
exports.deleteUserSubj = (req, res) => {
    const id = req.params.id
    console.log(`DELETE /users/api/${id}`)

    subject.findById(id, (err, subjects) => {

    		if (!subject) return res.status(404).send({message: 'Subject not saved'})

        subjects.remove( (err) => {
        	if (err) return res.status(500).send(err.message)
      		
      		res.status(200).jsonp({message: 'Subject deleted'})
        })
    })
}

// GET - Return the publickey from a specific user by id 
exports.getPublickeyUser = (req, res) => {
	var id = req.params.id
	console.log('POST /users/publickey/${id}')
	//console.log('GET /users/publickey/${id}')
	
	user.findById(id, {'_id':0,'token':0,'name':0,'lastname':0,'email':0,'typeuser':0,'salt':0,'__v':0,'hashed_password':0,'privatekey':0}, function(err, pbkey){
	 
		if (err) return res.send(500, err.message)
		res.status(200).jsonp(pbkey)
	  
	})
}

// GET - Return the publickey from a specific user by  email
exports.getPublickeyUserByEmail = (req, res) => {
	var email = req.params.email
	var s = '@uniovi.es';
	var e = email+s;
	console.log('POST /users/publickeyemail/${email}')
	//console.log('GET /users/publickey/${email}')
	
	user.findOne({email:e}, {'_id':0,'token':0,'name':0,'lastname':0,'email':0,'typeuser':0,'salt':0,'__v':0,'hashed_password':0,'privatekey':0}, function(err, pbkey){
	 
		if (err) return res.send(500, err.message)
		res.status(200).jsonp(pbkey)
	  
	})
}

// GET - Return the privatekey from a specific user (by id or email)
exports.getPrivatekeyUser = (req, res) => {
	var id = req.params.id
	// var email = req.params.email
	console.log('POST /users/privatekey/${id}')
	//console.log('GET /users/privatekey/${id}')
	
	user.findById(id, {'_id':0,'token':0,'name':0,'lastname':0,'email':0,'typeuser':0,'salt':0,'__v':0,'hashed_password':0,'publickey':0}, function(err, pvkey){
	 
		if (err) return res.send(500, err.message)
		res.status(200).jsonp(pvkey)
	  
	})
}

// GET - Return the email from a specific user by id 
exports.getEmailById = (req, res) => {
	var id = req.params.id
	// var email = req.params.email
	console.log('POST /users/getEmail/${id}')
	
	user.findById(id, {'_id':0,'token':0,'name':0,'lastname':0,'typeuser':0,'salt':0,'__v':0,'hashed_password':0,'publickey':0,'privatekey':0}, function(err, emayl){
	 
		if (err) return res.send(500, err.message)
		res.status(200).jsonp(emayl)
	  
	})
}
