// config/setStudent.js

var mongoose = require('mongoose');
var user = require('config/models');
var subject = require('config/subjects');

var users = mongoose.model('users');
var subjects = mongoose.model('subjects');


exports.saveSubj = function(id_user,subj,groupte,grouppa,grouppl,profname,profemail,callback) {


//subject.find({subj: subj},function(err,subjects){
	
//user.find({token: id_user},function(err,users){
	
//user.find({"_id":id}, function(err,users){
//subject.find({_id})	
//user.find({ users : users._id}, function(err, users) {
//user.findById(id, function(err, users) {
//user.find({ 'username': username,'email':email }, function(err, user) {
	/*console.log('idtoken: '+id_user);
	console.log('userid: '+users[0]._id);
	console.log('user: '+users);*/
	

user.find({token: id_user},function(err,users){
subject.find({subj: subj/*.replace(/\s/g, "")*/, User:users[0]._id},function(err,subjects){ // Buscamos asignatura con la id del usuario en linea
	

if(subjects.length == 0 && users.length != 0){
	
	
	//var newsubject = new subjects({
	var newsubject = new subject({	
	/*id : id,
    asigf1 : asigf1,
	asigf2 : asigf2,
	asigf3 : asigf3,
    asigf4 : asigf4,
	groupte : groupte,
    grouppa : grouppa,
    grouppl : grouppl, 
	profname : profname,
	profemail : profemail,
	optionsubj : optionsubj,
	user : user*/
	//id_user : id_user,
	subj : subj/*.replace(/\s/g, "")*/,
	groupte : groupte,
    grouppa : grouppa,
    grouppl : grouppl, 
	profname : profname,
	profemail : profemail,
	//Users : mongoose.Types.ObjectId(users[0]._id)
	//Users : [users[0]._id]
	User : users[0]._id
	
	});
	
	
		if (users.length != 0){
			newsubject.save(function (err) {
			callback("Subject Saved");
			});
		} else {
			callback("Can´t saved that subject");
		}
	
	
} // if
		
	else {
		
		if (users.length == 0){
			callback("Can´t saved that subject with this id_user");
		} else {
			
		
		//subject.find({subj: subj, id_user:id_user},function(err,subjec){
		//subject.find({subj: subj, User:users[0]._id},function(err,subjec){
			//var len = subjec.length;
			//if(len != 0){
				if (subjects.length !=0){
				
				callback("Subject already Saved");
 
			}/*else{

				//subject.update({ _id:subjects[0]._id},{$push:{"User":users[0]._id}},function(err,subje){
				subject.update({ _id:subjects[0]._id},{$addToSet:{"User":users[0]._id}},function(err,subje){
				callback({'res':"Subject Saved", 'ids':subjects[0].User});
				
				//console.log('subject: '+users[0]._id);
				//console.log('subject: '+subjects[0].Users);
				
				});
			}*/
		//});
		}	
	}






});			
});
}
