// config/subjects.js


//var async = require("async"); 
var mongoose = require('mongoose');
var Schema = mongoose.Schema;
var users = mongoose.model('users');


var subjectSchema = mongoose.Schema({
   
	/*asigf1 : [String],
	//asigf1 : String,
	//asigf1 : { type: String, enum: ['Álgebra Lineal ', 'Ampliación de Cálculo', 'Cálculo', 'Dispositivos Electrónicos y Fotónicos', 'Empresa', 'Estadística', 'Fundamentos de Informática', 'Mecánica y Termodinámica', 'Ondas y Electromagnetismo', 'Teoría de Circuitos']},
	asigf2 : [String],
	asigf3 : [String],
	asigf4 : [String],
	groupte : [String],
	grouppa : [String],
	grouppl : [String],
	profname : [String],
	profemail : [String],
	optionsubj : [String],
	user: { type: mongoose.Schema.Types.ObjectId, ref: "users" }
	
	//user: { type: Schema.ObjectId, ref: "users" },*/
	id_user : String,
	subj : String,
	groupte : String,
	grouppa : String,
	grouppl : [String],
	profname : String,
	profemail : String,
	User : {type : mongoose.Schema.Types.ObjectId, ref: "users"}
	//Users : [{type : mongoose.Schema.Types.ObjectId, ref: "users"}] 
	
	
});


//mongoose.connect('mongodb://localhost:27017/node-android');
module.exports = mongoose.model('subjects', subjectSchema);
